bus-mapping/src/circuit_input_builder/execution.rs

```diff
@@ -221,6 +221,14 @@ pub enum CopyDataType {
     /// scenario where we wish to accumulate the value (RLC) over all rows.
     /// This is used for Copy Lookup from SHA3 opcode verification.
     RlcAcc,
+    /// When the source for copy event is access-list addresses (EIP-2930) of tx-table.
+    TxAccessListAddresses,
+    /// When the destination for copy event is access-list addresses (EIP-2930) of rw-table.
+    RwAccessListAddresses,
+    /// When the source for copy event is access-list storage keys (EIP-2930) of tx-table.
+    TxAccessListStorageKeys,
+    /// When the destination for copy event is access-list storage keys (EIP-2930) of rw-table.
+    RwAccessListStorageKeys,
```

Comment:
I don't think this is the right approach to add 4 new enum variants. The copy circuit uses a `BinaryNumberGadget` to check for the data type. And we will need more bits to support these new types, hence will also increase the degree of the circuit.

Reply:
Since the `(src, dst)` tuples for these 2 cases are unique, i.e. `(TxAccessListAddresses, RwAccessListAddresses)` and `(TxAcessListStorageKeys, RwAccessListStorageKeys)` are the only 2 tuples. This means instead of 4 variants, we can also do with 2 variants and use the variant for both `source` and `destination`.

This would mean we will continue using the same number of bits in the copy circuit for the data type `BinaryNumberGadget`.

Reply:
Merge `CopyDataType` of access list to `AccessListAddresses` and `AccessListStorageKeys` (for both source and destination) in commit 097d3b8.